superset/views/core.py

```diff
@@ -1807,6 +1809,11 @@ def dashboard(self, dashboard_id):
                         "superset/request_access/?" f"dashboard_id={dash.id}&"
                     )
 
+        # Filter out unneeded fields from the datasource payload
+        datasources_payload = {}
```

Comment:
Could we make this a dictionary comprehension, 

```python
datasources_payload = {
    datasource.uid: datasource.data_for_slices(slices) for datasource, slices in datasources.items()
}
```

Reply:
done